From 1872db584d55a27ab08c8ed2c57cf2d3de984c65 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:58 -0400
Subject: [infiniband] ocrdma: Avoid posting DPP requests for RDMA READ

Message-id: <6a325f4c903546ecd4f62f1ec33e989490f738e8.1411571481.git.dledford@redhat.com>
Patchwork-id: 96611
O-Subject: [Patch RHEL7 23/48] RDMA/ocrdma: Avoid posting DPP requests for RDMA READ
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

Rollback the patch which allows DPP posting of RDMA READ since this
path is not yet stable in FW.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit f50f31e42fb772b62e209cf6ff5254b1bc02b756)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
index 3bbf201..bce4adf 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
@@ -2206,7 +2206,8 @@ int ocrdma_mbx_create_qp(struct ocrdma_qp *qp, struct ib_qp_init_attr *attrs,
     OCRDMA_CREATE_QP_REQ_RQ_CQID_MASK;
  qp->rq_cq = cq;
 
- if (pd->dpp_enabled && pd->num_dpp_qp) {
+ if (pd->dpp_enabled && attrs->cap.max_inline_data && pd->num_dpp_qp &&
+     (attrs->cap.max_inline_data <= dev->attr.max_inline_data)) {
   ocrdma_set_create_qp_dpp_cmd(cmd, pd, qp, enable_dpp_cq,
           dpp_cq_id);
  }
-- 
1.7.1